libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/AddGranteeSelect.tsx

```diff
@@ -1,35 +1,55 @@
 // (C) 2021 GoodData Corporation
 import React, { useCallback, useEffect, useMemo, useRef } from "react";
+import debounce from "debounce-promise";
 import { useIntl } from "react-intl";
-import Select, { ValueType, components as ReactSelectComponents, InputProps } from "react-select";
-import { IAddGranteeSelectProps, ISelectOption } from "./types";
-import { getGranteeItemTestId } from "./utils";
+import { ValueType } from "react-select";
+import AsyncSelect from "react-select/async";
+import { useBackendStrict, useWorkspaceStrict } from "@gooddata/sdk-ui";
+import { IAnalyticalBackend } from "@gooddata/sdk-backend-spi";
+import { areObjRefsEqual } from "@gooddata/sdk-model";
+
+import { IAddGranteeSelectProps, ISelectOption, isGranteeItem } from "./types";
+
+import {
+    EmptyRenderer,
+    GroupHeadingRenderer,
+    InputRendered,
+    LoadingMessageRenderer,
+    MenuListRendered,
+    NoOptionsMessageRenderer,
+    OptionRenderer,
+} from "./AsyncSelectComponents";
+import { loadGranteeOptionsPromise } from "./backend/loadGranteeOptionsPromise";
+
+const SEARCH_INTERVAL = 400;
 
 /**
  * @internal
  */
 export const AddGranteeSelect: React.FC<IAddGranteeSelectProps> = (props) => {
-    const { granteesOption, onSelectGrantee } = props;
+    const { appliedGrantees, onSelectGrantee } = props;
+
+    const backend: IAnalyticalBackend = useBackendStrict();
+    const workspace: string = useWorkspaceStrict();
 
     const intl = useIntl();
-    const selectRef = useRef(null);
+    const selectRef = useRef<AsyncSelect<ISelectOption, false>>(null);
 
     useEffect(() => {
-        selectRef.current.focus();
+        selectRef?.current.focus();
```

Comment:
Should probably be
```
selectRef.current?.focus();
```
because `selectRef` will always be defined, not so `selectRef.current`

Reply:
done